bulk-ingester/src/test/java/org/alfresco/hxi_connector/bulk_ingester/processor/mapper/AlfrescoPropertyMapperTest.java

```diff
@@ -0,0 +1,489 @@
+
+        alfrescoNode.setNodeProperties(
+                Set.of(
+                        createNodeProperty(description, stringValue(descriptionText))));
+
+        // when
+        var property = new AlfrescoPropertyMapper(alfrescoNode, description).performMapping();
+
+        // then
+        assertTrue(property.isPresent());
+        assertEquals(description, property.get().getKey());
+        assertEquals(descriptionText, property.get().getValue());
+    }
+
+    @Test
+    void shouldProcessNullProperties()
+    {
+        // given
+        AlfrescoNode alfrescoNode = new AlfrescoNode();
+
+        String description = "description";
+
+        alfrescoNode.setNodeProperties(
+                Set.of(
+                        createNodeProperty(description, nullValue())));
+
+        // when
+        var property = new AlfrescoPropertyMapper(alfrescoNode, description).performMapping();
+
+        // then
+        assertFalse(property.isPresent());
+    }
+
+    @Test
+    void shouldProcessBoolProperties()
+    {
+        // given
+        AlfrescoNode alfrescoNode = new AlfrescoNode();
+
+        String isIndexed = "isIndexed";
+
+        alfrescoNode.setNodeProperties(
+                Set.of(
+                        createNodeProperty(isIndexed, boolValue(true))));
+
+        // when
+        var property = new AlfrescoPropertyMapper(alfrescoNode, isIndexed).performMapping();
+
+        // then
+        assertTrue(property.isPresent());
+        assertEquals(isIndexed, property.get().getKey());
+        assertTrue((Boolean) property.get().getValue());
+    }
+
+    @Test
+    void shouldProcessLongProperties()
+    {
+        // given
+        AlfrescoNode alfrescoNode = new AlfrescoNode();
+
+        String legsCount = "legsCount";
+        long legsCountValue = 3;
+
+        alfrescoNode.setNodeProperties(
+                Set.of(
+                        createNodeProperty(legsCount, longValue(legsCountValue))));
+
+        // when
+        var property = new AlfrescoPropertyMapper(alfrescoNode, legsCount).performMapping();
+
+        // then
+        assertTrue(property.isPresent());
+        assertEquals(legsCount, property.get().getKey());
+        assertEquals(legsCountValue, property.get().getValue());
+    }
+
+    @Test
+    void shouldProcessFloatProperties()
+    {
+        // given
+        AlfrescoNode alfrescoNode = new AlfrescoNode();
+
+        String legLength = "legLength";
+        float legLengthValue = 3.5f;
+
+        alfrescoNode.setNodeProperties(
+                Set.of(
+                        createNodeProperty(legLength, floatValue(legLengthValue))));
+
+        // when
+        var property = new AlfrescoPropertyMapper(alfrescoNode, legLength).performMapping();
+
+        // then
+        assertTrue(property.isPresent());
+        assertEquals(legLength, property.get().getKey());
+        assertEquals(legLengthValue, property.get().getValue());
```

Comment:
This is fine because we know we're just passing the float values through the tested method (same for the next test with doubles).